### ISSUE TITLE
Update Logstash to at least 7.16.3

### ISSUE THREAD

**Issue by ccostino** [cloud-gov/logsearch-boshrelease#132]
In order to stay ahead of security updates and patches, we need to update Logstash to at least version [7.16.3](https://www.elastic.co/guide/en/logstash/7.17/logstash-7-16-3.html), primarily to make sure we are running Log4j 2.17.1 or later.

Please note that the latest release in the 7.x series is [7.17.1](https://www.elastic.co/guide/en/logstash/7.17/logstash-7-17-1.html); the absolute latest release of Logstash is [8.1.0](https://www.elastic.co/guide/en/logstash/current/logstash-8-1-0.html), which we probably can't move to just yet.

## Acceptance Criteria
* [ ] Logstash is updated to at least version 7.16.3, and Log4j is updated to at least 2.17.1.

--- 

## Security Considerations
- Helps us stay up-to-date and secure by applying the latest security patches
- Ensures that we stay on top of all Log4j updates
- Note that we already have security provisions and mitigations in place for the main issues with the Log4j vulnerabilities that were reported back in 12/2021

## Implementation sketch
- See PR #113 to see how this was done previously

**Comment by markdboyd** [comment]
PR: https://github.com/cloud-gov/logsearch-boshrelease/pull/135

**Comment by markdboyd** [comment]
https://github.com/cloud-gov/logsearch-boshrelease/pull/137

**Comment by markdboyd** [comment]
https://github.com/cloud-gov/logsearch-boshrelease/pull/138

**Comment by markdboyd** [comment]
I ran into problems on this ticket of my own making.

In https://github.com/cloud-gov/logsearch-boshrelease/pull/136/files, I updated our version of Elasticsearch in the BOSH release to go from 7.9.3 to 7.17.5. While the deploy of the next BOSH release with Elasticsearch 7.17.5, I initially thought it was incompatible with Kibana 7.9.3. So I rolled ES back to 7.9.3 in https://github.com/cloud-gov/logsearch-boshrelease/pull/138.

But the next BOSH release deployment with ES 7.9.3 was failing because Elasticsearch does not support downgrading from 7.17.5 to 7.9.3. The problem seems to be that data was stored in the index using a version of Lucene that exists in ES 7.17.5 but not 7.9.3 and this data cannot be loaded once you downgrade from 7.17.5 to 7.9.3. The only solution we determined was to completely delete the BOSH deployment for Logsearch in dev, redeploy it, and then [follow our docs to restore data from S3](https://cloud.gov/docs/ops/runbook/troubleshooting-logsearch/#reindexing-whole-days-from-s3). 

Unfortunately, before these downgrade issues were noticed, the update to Elasticsearch 7.17.5 had already deployed to logs stage, logs-platform stage, and logs-platform prod. So at minimum we're going to have to do the delete and restore process in stage.

@pburkholder mentioned that he would like to use the stage restore process as our team quarterly backup exercise.

As an aside, it turns out that ES 7.17.5 and Kibana 7.9.3 do seem compatible.

To do:

- [ ] Delete logsearch-platform BOSH deployment in dev, re-deploy it, and then restore data from S3 (how far back do we need to go in restoring data?)
- [ ] Delete logsearch BOSH deployment in stage, re-deploy it, and then restore data from S3 (how far back do we need to go in restoring data?)
- [ ] Delete logsearch-platform BOSH deployment in stage, re-deploy it, and then restore data from S3 (how far back do we need to go in restoring data?)
- [ ] Split logsearch-boshrelease to use different versions of ES for logs and logs-platform

**Comment by markdboyd** [comment]
https://github.com/cloud-gov/cg-site/pull/2158

**Comment by pburkholder** [comment]
Did we get things to a state where Logstash is updated (but the rest of the stack is doing its own thing)?

**Comment by markdboyd** [comment]
@pburkholder Where we are now is that Logsearch for tenants is only ES version 7.9.3 and Kibana version 7.9.3. But Logsearch platform is on ES version 7.17.5 and Kibana 7.17.5, since we determined it was safe to use those versions with upgraded licenses for the platform.

We've only tried the deployment of ES/Kibana 7.17.5 in logs-platform dev and it is currently failing, but we made a few changes to the BOSH release so just need to figure out what the problem is and fix it.

Does that answer your question?
